### PR TITLE
Updates to Readme.md

### PR DESCRIPTION
**Pull request by powderluv**
Clean up the Readme.md to make it easy and focussed on checkout and build. 

Will update Demos next.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,68 +30,50 @@ We have few paths to lower down to the Torch MLIR Dialect.
...
-# Run write_env_file.sh to emit a .env file with needed
-# PYTHONPATH setup.
-./build_tools/write_env_file.sh
-source .env
+## Setup your Python Environment
 
+```
+python3 -m venv mlir_venv
+source mlir_venv/bin/activate
+pip3 install --upgrade pip #Some older pip installs may not be able to handle the recent PyTorch deps
```
Comment (stellaraccident) [inline README.md:46]: I usually just use "python" and "pip" after venv activation. The "3" variants are not actually available in all platforms.

File: README.md [hunk 2]
```diff
@@ -30,68 +30,50 @@ We have few paths to lower down to the Torch MLIR Dialect.
...
+# Install latest PyTorch nightlies
+pip3 install --pre torch torchvision pybind11 -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
```
Comment (stellaraccident) [inline README.md:48]: Prefer `python -m pip`. There is less environmental foo that can go wrong with it.

File: README.md [hunk 3]
```diff
@@ -30,68 +30,50 @@ We have few paths to lower down to the Torch MLIR Dialect.
...
 ```
 
+## Build 
+```
+cmake -GNinja -Bbuild \
+  -DCMAKE_C_COMPILER=clang-13 \
```
Comment (stellaraccident) [inline README.md:54]: Is forcing clang (and version 13) load bearing? (That's what I think when I see such specifics).

File: README.md [hunk 4]
```diff
@@ -30,68 +30,50 @@ We have few paths to lower down to the Torch MLIR Dialect.
...
+  -DLLVM_ENABLE_PROJECTS=mlir \
+  -DLLVM_EXTERNAL_PROJECTS=torch-mlir \
+  -DLLVM_EXTERNAL_TORCH_MLIR_SOURCE_DIR=`pwd` \
+  -DMLIR_ENABLE_BINDINGS_PYTHON=ON \
+  -DLLVM_TARGETS_TO_BUILD=host \
+  external/llvm-project/llvm
+
+# Additional quality of life CMake flags:
+# Enable ccache:
+#   -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
```
Comment (stellaraccident) [inline README.md:65]: Probably want lld in here. It reduces some of the link jobs from minutes to seconds, especially in debug builds.

File: README.md [hunk 5]
```diff
@@ -127,37 +108,25 @@ TODO
 
 TODO
 
-### Additional TorchScript end-to-end tests with heavy dependencies
 
-Some of the Torchscript end-to-end tests require additional dependencies which
-don't make sense to include as part of the default torch-mlir setup.
```
Comment (silvasean) [inline README.md:133]: can you move this into a comment in build_tools/torchscript_e2e_heavydep_tests/generate_serialized_tests.sh -- it's the only load-bearing documentation for this feature, so I don't want to outright delete it.

File: README.md [hunk 6]
```diff
@@ -102,7 +84,7 @@ Running execution (end-to-end) tests:
 ./tools/torchscript_e2e_test.sh --filter Conv2d --verbose
 ```
 
-Standalone script:
+Standalone script to generate and run a Resnet18 model:
```
Comment (silvasean) [inline README.md:87]: nit: ResNet18

**Review by silvasean — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (silvasean) [reply README.md:133]: Actually build_tools/torchscript_e2e_heavydep_tests/README.md would be better

File: README.md (hunk 3 above)
Comment (powderluv) [reply README.md:54]: no will remove version. Was copying form the original CC export but will assume we still want to use clang / clang++

File: README.md (hunk 4 above)
Comment (powderluv) [reply README.md:65]: done

File: README.md (hunk 6 above)
Comment (powderluv) [reply README.md:87]: done

File: README.md (hunk 5 above)
Comment (powderluv) [reply README.md:133]: Will do in a follow on commit

**Pull request merged**